src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -145,22 +152,55 @@ def generate_json_report(self):
     def _populate_core_stats(self, area):
         if area.uuid not in self.flattened_area_core_stats_dict:
             self.flattened_area_core_stats_dict[area.uuid] = {}
-        if self.current_market_unix is not None and \
-                self.current_market_unix not in self.flattened_area_core_stats_dict[area.uuid]:
-            core_stats_dict = {'bids': [], 'offers': [], 'trades': []}
-            if hasattr(area.current_market, 'offer_history'):
-                for offer in area.current_market.offer_history:
-                    core_stats_dict['offers'].append(offer.serializable_dict())
-            if hasattr(area.current_market, 'bid_history'):
-                for bid in area.current_market.bid_history:
-                    core_stats_dict['bids'].append(bid.serializable_dict())
-            if hasattr(area.current_market, 'trades'):
-                for trade in area.current_market.trades:
-                    core_stats_dict['trades'].append(trade.serializable_dict())
-
-            self.flattened_area_core_stats_dict[area.uuid] = {}
-            self.flattened_area_core_stats_dict[area.uuid][self.current_market_unix] = \
-                core_stats_dict
+        if self.current_market == "":
+            return
+        core_stats_dict = {'bids': [], 'offers': [], 'trades': []}
+        if hasattr(area.current_market, 'offer_history'):
+            for offer in area.current_market.offer_history:
+                core_stats_dict['offers'].append(offer.serializable_dict())
+        if hasattr(area.current_market, 'bid_history'):
+            for bid in area.current_market.bid_history:
+                core_stats_dict['bids'].append(bid.serializable_dict())
+        if hasattr(area.current_market, 'trades'):
+            for trade in area.current_market.trades:
+                core_stats_dict['trades'].append(trade.serializable_dict())
+
+        if isinstance(area.strategy, PVStrategy) or \
+                isinstance(area.strategy, PVUserProfileStrategy) or \
+                isinstance(area.strategy, PVPredefinedStrategy):
```

Comment:
No need for these 2, they inherit from the PVStrategy therefore isinstance should detect them.